src/loaders/compute_tools/microtrait/microtrait_util.py

```diff
@@ -0,0 +1,68 @@
+def decode_rule_boolean(
+        rule_name: str,
+        rule_name_boolean_dict: dict[str, str],
+        depth: int = 5,
+) -> str:
+    """
+    Given a rule name and a dictionary of rule names and their corresponding boolean expressions,
+    return the boolean expression at the gene level for the given rule name.
+    """
+
+    if rule_name not in rule_name_boolean_dict:
+        return ''
+
+    rule_boolean = rule_name_boolean_dict[rule_name]
+
+    # Replace any existing rule names with their corresponding boolean expressions
+    for k, v in rule_name_boolean_dict.items():
+
+        # special case when rule boolean is another rule
+        if f"('{rule_name}')" in rule_boolean:
+            rule_boolean = rule_boolean.replace(f"('{rule_name}')", f"((((((('{rule_name}')))))))")
+            return rule_boolean
```

Comment:
I did this because in the rules_asserted table, when rule bool itself is also a rule name, it formats like this: '('motA') -> (((((((1)))))))`, `('mntH') -> (((((((0)))))))`. I wasn't able to find an explanation from the microtrait codebase. So I have to do it this way to match what the rules_asserted table shows.